internal/auth/options.go

```diff
@@ -388,19 +393,14 @@ func SetProvider(provider providers.Provider) func(*Authenticator) error {
 
 // AssignStatsdClient is function that takes in an Options struct and assigns a statsd client
 // to the proxy and provider.
-func AssignStatsdClient(opts *Options) func(*Authenticator) error {
+func AssignStatsdClient(statsdClient *statsd.Client) func(*Authenticator) error {
 	return func(proxy *Authenticator) error {
```

Comment:
unrelated nit: can we change this variable name to be authenticator rather than proxy?

Reply:
Done.